xbmc/cores/AudioEngine/Sinks/AESinkDARWINOSX.cpp

```diff
@@ -93,13 +93,56 @@ static CADeviceList GetDevices()
   return list;
 }
 
+//---------- DEVICE CHANGED EVENT HANDLING
+// when device list change callbacks are received
+// inform the ae engine this amount of time after
+// the last callback was received (bundle multiple callbacks)
+// into one final call
+#define DEVICE_LIST_CHANGED_TIMER_MS 1500
+
+CDeviceChangeListTimeout::CDeviceChangeListTimeout()
+{
+  m_pDeviceListChangedTimer = new CTimer(this);
```

Comment:
Yep you are right - that would be the use case where the audio device the sink is using gets removed (in my case i turn off the bluetooth speaker i play audio through for example). You are right that AE gets a timeout in addpackets, destroys the sink and reenumerates the devices. The timer is still active and fires the delayed device changed event. Well this is assuming, that the OS is already aware of the lost device and AE really gets the up to date list (where the bluetooth speaker is already removed).
As i said in the ticket ... at least some use cases to think off and some assumptions to make. Maybe i simply let it be as is. It works and not much to gain from this anyway i think.